### PR TITLE
[DUOS-566][risk=no] Handle missing ontology terms better

### PR DESCRIPTION
**Pull request by rushtong**
## Addresses
https://broadinstitute.atlassian.net/browse/DUOS-566

## Changes
* Catch errors with purpose terms before continuing on with other algorithm checks

### CODE REVIEW

File: src/test/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatcherTest.java [hunk 1]
```diff
@@ -271,11 +270,7 @@ private void assertNegative(DataUse purpose, DataUse dataset) {
     private ImmutablePair<Boolean, List<String>> matchPurposeAndDataset(DataUse purpose, DataUse dataset) {
         DataUseMatcher matcher = new DataUseMatcher();
         matcher.setAutocompleteService(autocompleteService);
-        try {
```
Comment (Marianie-Alphonse) [inline src/test/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatcherTest.java:274]: Why did you remove the try-catch error handling here? Does it get handled in another place?

Comment (rushtong) [reply src/test/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatcherTest.java:274]: Yes, exactly, line 43 of `DataUseMatcher` is the method signature which no longer throws an exception. Inside the method, we handle the error case.

Comment (Marianie-Alphonse) [reply src/test/java/org/broadinstitute/dsde/consent/ontology/datause/DataUseMatcherTest.java:274]: sounds good!

**Review by Marianie-Alphonse — APPROVED** [review]
